cl/search/management/commands/cl_compute_token_count.py

```diff
@@ -0,0 +1,205 @@
+
+
+def compute_avg_from_list(array: list[float]) -> float:
+    """
+    Computes the average of a list of numbers.
+
+    Args:
+        array (list[float]): A list containing numeric values.
+
+    Returns:
+        float: The average of the numbers in the list as a floating-point
+         value. If the list is empty, returns 0.0.
+    """
+    return sum(array) / len(array)
+
+
+class Command(VerboseCommand):
+    help = "Compute token count for Recap Documents and Caselaw."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--percentage",
+            type=float,
+            default=0.1,
+            help="specifies the proportion of the table to be sampled",
+        )
+
+    def handle(self, *args, **options):
+        percentage = options["percentage"]
+        rd_queryset = get_recap_random_dataset(percentage)
+
+        token_count = []
+        tokens_per_page = []
+        words_per_page = []
+        self.stdout.write("Starting to retrieve the random RECAP dataset.")
+        for document in rd_queryset.iterator():
+            count = get_token_count_from_string(document.plain_text)
+            token_count.append(count)
+            tokens_per_page.append(count / document.page_count)
+            word_count = len(document.plain_text.split())
+            words_per_page.append(ceil(word_count / document.page_count))
+
+        self.stdout.write("Computing averages.")
+        sample_size = len(token_count)
+        avg_tokens_per_doc = compute_avg_from_list(token_count)
+        avg_words_per_page = compute_avg_from_list(words_per_page)
+        avg_tokens_per_page = compute_avg_from_list(tokens_per_page)
+
+        self.stdout.write(
+            "Counting the total number of document in the Archive."
+        )
+        total_recap_documents = RECAPDocument.objects.all().count()
+        total_token_in_recap = avg_tokens_per_doc * total_recap_documents
+
+        self.stdout.write(f"Size of the dataset: {sample_size}")
+        self.stdout.write(f"Average tokens per document: {avg_tokens_per_doc}")
+        self.stdout.write(f"Average words per page: {avg_words_per_page}")
+        self.stdout.write(f"Average tokens per page: {avg_tokens_per_page}")
+        self.stdout.write("-" * 20)
+        self.stdout.write(
+            f"Total number of recap documents: {total_recap_documents}"
+        )
+        self.stdout.write(
+            f"The sample represents {sample_size/total_recap_documents:.3%} of the Archive"
+        )
+        self.stdout.write(
+            f"Total number of tokens in the recap archive: {intword(total_token_in_recap)}"
+        )
+
+        opinion_queryset = get_opinions_random_dataset(percentage)
+        self.stdout.write("Starting to retrieve the random Opinion dataset.")
+        for opinion in opinion_queryset.iterator():
+            text = get_clean_opinion_text(opinion)
+            count = get_token_count_from_string(text)
+            token_count.append(count)
```

Comment:
Don't you need to clear the `token_count` variable before you start appending to it again?

Reply:
you're right!. Thanks for catching that.